models/users.js

```diff
@@ -117,15 +117,23 @@ const addOrUpdate = async (userData, userId = null) => {
     }
 
     // userId is null, Add or Update user
-    let user;
+    let user = null;
+
     if (userData.github_user_id) {
       user = await userModel.where("github_user_id", "==", userData.github_user_id).limit(1).get();
     }
-    if (!user || (user && user.empty)) {
+
+    if (userData.github_id && (!user || user.empty)) {
       user = await userModel.where("github_id", "==", userData.github_id).limit(1).get();
     }
+
+    if (userData.email && (!user || user.empty)) {
+      user = await userModel.where("email", "==", userData.email).limit(1).get();
+    }
+
     if (user && !user.empty && user.docs !== null) {
-      await userModel.doc(user.docs[0].id).set({ ...userData, updated_at: Date.now() }, { merge: true });
+      const { created_at: createdAt, ...updatedUserData } = userData;
+      await userModel.doc(user.docs[0].id).set({ ...updatedUserData, updated_at: Date.now() }, { merge: true });
```

Comment:
Currently, we're performing sequential Firestore queries. so this would lead to slower response times as each query waits for the previous one to complete. we should consider using `Promise.all()` to run these queries in parallel.

Reply:
I see your point about early termination when a match is found. However, in the worst case (no match found), we still end up running all queries sequentially. The actual performance impact would depend on our typical use cases. Let's monitor query performance in production and revisit this if we see latency issues.